runtime/vm/JVMImage.cpp

```diff
@@ -142,6 +261,34 @@ JVMImage::freeSubAllocatedMemory(void* address)
 	omrthread_monitor_exit(_jvmImageMonitor);
 }
 
+void
+JVMImage::registerEntryInTable(ImageTableHeader *table, UDATA entry)
+{
+	Trc_VM_RegisterInTable_Entry(table, table->currentSize, *(WSRP_GET(table->tableTail, UDATA*)), entry);
+
+	// table is not large enough and needs to be reallocated
+	if (table->currentSize >= table->tableSize) {
+		reallocateTable(table, table->tableSize * 2); //TODO: Introduce error handling for table reallocation
+	}
+
+	UDATA *head = WSRP_GET(table->tableHead, UDATA*);
+	UDATA *tail = WSRP_GET(table->tableTail, UDATA*);
+
+	// initial state of every table
+	if (table->currentSize == 0) {
+		*(head) = entry;
+	}
+	else {
+		tail += 1;
+		WSRP_SET(table->tableTail, tail);
+	}
+
+	*(tail) = entry;
```

Comment:
you have a double assignment here if `table->currentSize == 0` is true

Reply:
is `table->tableTail` ever different from `head + table->currentSize`?

Reply:
no its set to that. Well technically tail = head + currentSize - 1 unless table is empty in which case tail = head + currentSize(0).

Reply:
Ill change conditional to only check if currentSize != 0,  no need for *head = entry since head = tail at start